### PR TITLE
Crypto hamsters

### PR DESCRIPTION
**Pull request by alexeitcekov**

### CODE REVIEW

File: contracts/crypto-hamster/crypto-hamsters.aes [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            id = state.index,
+            name = _name,
+            dna = _dna}
+
+        put(state{nameHamster[_name] = newHamster})
+        put(state{index = (state.index + 1)})
+
+    private function generateRandomDNA() : int =
+        // I want to create a hash of the input str like: let rand = int(str_hash))
+        Chain.block_hash(Chain.block_height - 1) - Chain.timestamp
```
Comment (maptuhec) [inline contracts/crypto-hamster/crypto-hamsters.aes]: why not + Chain.timestamp?

File: contracts/crypto-hamster/crypto-hamsters.aes [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+// pragma solidity ^0.4.25;
+// contract CryptoHamster {
+
+//     event NewHamster(uint hamsterId, string name, uint dna);
+
+//     uint dnaDigits = 16;
+//     uint dnaModulus = 10 ** dnaDigits;
+
+//     struct Hamster {
+//         string name;
```
Comment (maptuhec) [inline contracts/crypto-hamster/crypto-hamsters.aes]: remove commented code

File: contracts/crypto-hamster/crypto-hamsters.aes (hunk 1 above)
Comment (Perseverance) [reply contracts/crypto-hamster/crypto-hamsters.aes]: Lets add `+id`

File: contracts/crypto-hamster/crypto-hamsters.aes [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        { index = 1,
+          nameHamster = {}}
+
+    public function createHamster(hamsterName: string) =
+        require(!isNameExist(hamsterName), "Name is already taken")
+
+        // let randDna = generateRandomDNA(name)
+        createHamsterByNameDNA(hamsterName, generateRandomDNA())
+
+    public function isNameExist(name: string) : bool =
```
Comment (Perseverance) [inline contracts/crypto-hamster/crypto-hamsters.aes]: I think we should name this just `nameExists`

File: contracts/crypto-hamster/crypto-hamsters.aes [hunk 4]
```diff
@@ -0,0 +1,83 @@
+contract CryptoHamster =
+
+    record state = {
+        index : int,
+        nameHamster : map(string, hamster)}
```
Comment (Perseverance) [inline contracts/crypto-hamster/crypto-hamsters.aes]: Lets rename this to just `hamsters`

**Pull request merged**